openeo/rest/http.py

```diff
@@ -0,0 +1,39 @@
+from typing import Set
+
+import requests
+import requests.adapters
+
+MAX_RETRIES = 3
```

Comment:
default of max 3 retries might be a bit low in practice in combination with default backoff_factor: last attempt will be roughly just be 4 seconds after first attempt, which might be too quick to stop trying